### PR TITLE
Enhancements and bug fixes for the ecs_service module

### PR DESCRIPTION
**Pull request by rmorlok**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

cloud/amazon/ecs_service
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /Users/rmorlok/Dropbox/ryanroot/ryan-work/Contently/ContentlyDev/ansible/ansible.cfg
  configured module search path = ['/Users/rmorlok/Development/ansible-modules-extras']
```
##### SUMMARY

Output of command did not change except in error condition in which a stack trace is replaced with a descriptive error.

Added new parameters `deployment_maximum_percent` and
`deployment_minimum_healthy_percent` to allow ecs to potentially reduce or
increase the number of running tasks during a deployment.

Fixed issue where the module was unable to detect if an existing service
matched when the family name was used.

Fixed error checking for `task_definition` and `desired_count` parameters.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @rmorlok for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rmorlok** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rmorlok. To the current maintainers, @Java1Guy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rmorlok** [comment]
@Java1Guy same thing on this one

**Comment by rmorlok** [comment]
@Java1Guy pinging this PR again

**Comment by rmorlok** [comment]
@gregdek Same thing on this PR, not seeing any movement. Anyone else who could look at this?

**Comment by gregdek** [comment]
@rmorlok Let's give him a week and if no response ping me again. (Fair warning: I may ask you to become a co-maintainer at that point.)

File: cloud/amazon/ecs_service.py [hunk 1]
```diff
@@ -292,13 +316,15 @@ def main():
...
-        load_balancers=dict(required=False, type='list' ),
-        desired_count=dict(required=False, type='int' ),
-        client_token=dict(required=False, type='str' ),
+        state=dict(required=True, choices=['present', 'absent', 'deleting']),
+        name=dict(required=True, type='str'),
+        cluster=dict(required=False, type='str'),
+        task_definition=dict(required=False, type='str'),
+        load_balancers=dict(required=False, type='list'),
+        desired_count=dict(required=False, type='int'),
+        deployment_maximum_percent=dict(required=False, type='int'),
```
Comment (jamesongithub) [inline cloud/amazon/ecs_service.py]: keep these consistent? deployment_maximum_percent vs deployment_minimum_healthy_percent

File: cloud/amazon/ecs_service.py [hunk 2]
```diff
@@ -238,7 +247,8 @@ def describe_service(self, cluster_name, service_name):
         raise StandardError("Unknown problem describing service %s." % service_name)
 
     def is_matching_service(self, expected, existing):
```
Comment (jamesongithub) [inline cloud/amazon/ecs_service.py:249]: also compare `deploymentConfiguration`?

**Comment by rmorlok** [comment]
@gregdek still haven't heard from @Java1Guy on this PR either. Could someone else please take a look?

**Comment by gregdek** [comment]
OK, so looks like @Java1Guy is MIA on this one. @rmorlok willing to become a co-maintainer of this module?

**Comment by rmorlok** [comment]
Sure, no problem

**Comment by rmorlok** [comment]
@gregdek can I be added as a maintainer on this module as well, please?

**Comment by gregdek** [comment]
Done -- and I've given you maintainership on all the other ecs_\* modules as well.

**Comment by rmorlok** [comment]
shipit

**Comment by kkarimi** [comment]
Can this be merged yet?

**Comment by rmorlok** [comment]
@gregdek The Ansibull-bot didn't pick up my `shipit`.

I don't think your change regarding maintainers of the ecs_\* modules went through. (https://github.com/ansible/ansibullbot/blob/master/MAINTAINERS-EXTRAS.txt#L23)

**Comment by gregdek** [comment]
Sorry @rmorlok -- you were only listed as maintainer of one of the ecs modules. I've added you as a maintainer for each of them. Kicking the bot, it should pick up your shipit.

**Comment by gregdek** [comment]
Weird, dunno if that change was reverted or what.

**Comment by gregdek** [comment]
Thanks @rmorlok. To the current maintainers, @Java1Guy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Kicking again.

**Comment by gregdek** [comment]
Thanks again to @rmorlok. This module is going into community review.

We notice that you are one of the maintainers (@Java1Guy, @rmorlok) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rmorlok** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @rmorlok. Since you are one of the maintainers (@Java1Guy, @rmorlok) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by simplesteph** [comment]
@rmorlok just saw this... I had a slightly different approach on my PR, which matches boto API more closely... check out https://github.com/ansible/ansible-modules-extras/pull/3135 and let me know your thoughts. cc @zacblazic

**Comment by bcoca** [comment]
removed shipit as @jamesongithub's comment seems unresolved

I would also look and comment on @simplesteph's competing request

**Comment by gregdek** [comment]
Thanks again to @rmorlok for this PR. This PR was reviewed by an Ansible member. Marking for inclusion.

However, we would still appreciate if the maintainer(s) @Java1Guy, @rmorlok could look over it.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**